### PR TITLE
Collection avatar component

### LINKED ISSUE #17463: Create collection avatar component
Designs - 
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=28941-78785&mode=dev

Screenshot 2023-09-30 at 21 10 55

Acceptance Criteria
component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

### PR DESCRIPTION
**Pull request by vkjr**
fixes #17463

### Summary

Collection avatar added to the list of quo components

Screenshot 2023-09-30 at 21 08 30

Design: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=28941-78785&mode=dev

### Design notes:
Quo component can be found in "avatars" section of the app.
Screenshot 2023-10-02 at 11 32 46

status: ready

### CODE REVIEW

File: src/quo2/components/avatars/collection_avatar/component_spec.cljs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(ns quo2.components.avatars.collection-avatar.component-spec
+  (:require [quo2.components.avatars.collection-avatar.view :as collection-avatar]
+            [test-helpers.component :as h]))
+
+(defonce bored-ape (js/require "../resources/images/mock2/bored-ape.png"))
```
Comment (J-Son89) [inline src/quo2/components/avatars/collection_avatar/component_spec.cljs]: is it possible to get this from the resources file instead? 👍

File: src/quo2/components/avatars/collection_avatar/view.cljs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+(ns quo2.components.avatars.collection-avatar.view
+  (:require [quo2.components.avatars.collection-avatar.style :as style]
+            [quo2.theme :as quo.theme]
+            [react-native.fast-image :as fast-image]
+  ))
```
Comment (J-Son89) [inline src/quo2/components/avatars/collection_avatar/view.cljs]: nit: formatting is strange here

File: src/quo2/core.cljs [hunk 3]
```diff
@@ -149,6 +150,7 @@
 (def icon-avatar quo2.components.avatars.icon-avatar/icon-avatar)
 (def user-avatar quo2.components.avatars.user-avatar.view/user-avatar)
 (def wallet-user-avatar quo2.components.avatars.wallet-user-avatar/wallet-user-avatar)
+(def collection-avatar quo2.components.avatars.collection-avatar.view/view)
```
Comment (J-Son89) [inline src/quo2/core.cljs]: can we keep this list alpha sorted too 🙏

**Review by J-Son89 — APPROVED** [review]
few small comments, book looks good otherwise 🚀

File: src/quo2/components/avatars/collection_avatar/component_spec.cljs (hunk 1 above)
Comment (vkjr) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: @J-Son89, do you mean not from the `mock` folder?

Comment (J-Son89) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: no that's fine. I mean that we have a quo2.resources file so can we use that instead? and do the `resources/get-mock-image` approach?

Comment (vkjr) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: @J-Son89 ah, okay)

Comment (vkjr) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: @J-Son89, just remembered why it is this way.`resources/get-mock-image` located in `status-im2`, whereas component is in quo2. And our linter fails with warnings when we reference status-im2 from quo2.

Comment (J-Son89) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: Oh we have another file like that within quo2 for this reason though

**Comment by vkjr** [comment]
@Francesca-G, could you please take a look at this one?

File: src/quo2/components/avatars/collection_avatar/component_spec.cljs (hunk 1 above)
Comment (vkjr) [reply src/quo2/components/avatars/collection_avatar/component_spec.cljs]: @J-Son89, thanks, missed it!)

**Review by Francesca-G — APPROVED** [review]
Looks good ✨

**Pull request merged**